examples/text_to_image/train_text_to_image.py

```diff
@@ -697,6 +721,51 @@ def collate_fn(examples):
             if global_step >= args.max_train_steps:
                 break
 
+        if accelerator.is_main_process:
+            if args.validation_prompt is not None and epoch % args.validation_epochs == 0:
+                logger.info(
+                    f"Running validation... \n Generating {args.num_validation_images} images with prompt:"
+                    f" {args.validation_prompt}."
+                )
+                # create pipeline
+                safety_checker = StableDiffusionSafetyChecker.from_pretrained(
+                    args.pretrained_model_name_or_path, subfolder="safety_checker", revision=args.non_ema_revision
+                )
```

Comment:
`StableDiffusionPipeline.from_pretrained` should automatically load `safety_checker` when available,  is there any reason we need to load it here explicitly?

Reply:
Also, I'm not sure what difference it makes when we load `safety_checker` separately like this, `StableDiffusionPipeline.from_pretrained` does pretty much the same thing.